v2/tools/generator/internal/codegen/pipeline/handle_float_values.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// HandleFloatValuesStageID is the unique identifier of this stage
+const (
+	HandleFloatValuesStageID = "handleFloatValues"
+)
+
+func HandleFloatValues() *Stage {
+	stage := NewStage(
+		HandleFloatValuesStageID,
+		"Transform Float type values to Integer types",
```

Comment:
We're doing this because of a limitation in `controller-gen` so we should mention that.

```suggestion
		"Transform validated float type values to validated integer types for compatibility with controller-gen",
```